src/test/java/pl/debuguj/parkingspacessystem/dao/ParkingSpaceDaoMockImplTest.java

```diff
@@ -0,0 +1,189 @@
+        parkingSpace = new ParkingSpace(registrationNo, beginDate, endDate);
+
+    }
+
+    @After
+    public void after(){
+        parkingSpaceDao.removeAll();
+    }
+
+    @Test
+    public void testCreatingNewParkingSpace() throws Exception {
+
+        parkingSpaceDao.create(parkingSpace);
+
+        Optional<ParkingSpace> psFromDao = parkingSpaceDao.findByRegistrationNo(parkingSpace.getCarRegistrationNumber());
+
+        assertNotNull(psFromDao);
+
+        assertEquals(parkingSpace, psFromDao.get());
+        assertEquals(parkingSpace.getBeginTime(),psFromDao.get().getBeginTime());
+        assertEquals(parkingSpace.getDriverType(),psFromDao.get().getDriverType());
+        assertEquals(parkingSpace.getEndTime(),psFromDao.get().getEndTime());
+    }
+
+    @Test(expected = CarRegisteredInSystemException.class)
+    public void testCreatingNewParkingSpaceException() throws Exception {
+
+        parkingSpaceDao.create(parkingSpace);
+
+        parkingSpaceDao.create(parkingSpace);
+    }
+    @Test
+    public void testFindingParkingSpaceByRegistrationNo() throws Exception {
+
+        parkingSpaceDao.create(parkingSpace);
+
+        Optional<ParkingSpace> psFromDao = parkingSpaceDao.findByRegistrationNo(parkingSpace.getCarRegistrationNumber());
+
+        assertTrue(psFromDao.isPresent());
+
+        String registrationNo = "997755";
+
+        psFromDao = parkingSpaceDao.findByRegistrationNo(registrationNo);
+
+        assertFalse(psFromDao.isPresent());
+    }
+
+    @Test
+    public void testGettingAllParkingSpaces() throws Exception {
+        createTestItems();
+
+        int size = parkingSpaceDao.getAll().size();
+
+        assertEquals("Should return 5 number od object", 5, size);
+    }   
+
+    @Test
+    public void testUpdatingParkingSpaceEndTimeToNewCorrectEndTime() throws Exception  {
+
+        parkingSpaceDao.create(parkingSpace);
+
+        Date newCorrectDate = timeDateFormat.parse("2017-10-14 12:15:48");
+        parkingSpaceDao.updateEndTime(parkingSpace.getCarRegistrationNumber(), newCorrectDate);
+
+        Optional<ParkingSpace> psFromDao = parkingSpaceDao.findByRegistrationNo(parkingSpace.getCarRegistrationNumber());
+
+        assertTrue(psFromDao.isPresent());
+        assertEquals(parkingSpace, psFromDao.get());
+        assertEquals(parkingSpace.getBeginTime(), psFromDao.get().getBeginTime());
+        assertEquals(parkingSpace.getDriverType(), psFromDao.get().getDriverType());
+        assertEquals(newCorrectDate, psFromDao.get().getEndTime());
+    }
+
+    @Test(expected = IncorrectEndDateException.class)
+    public void testUpdatingParkingSpaceEndTimeToNewIncorrectEndTime() throws Exception  {
+
+        parkingSpaceDao.create(parkingSpace);
+
+        Date newIncorrectDate = timeDateFormat.parse("2017-10-14 10:15:48");
+
+        parkingSpaceDao.updateEndTime(parkingSpace.getCarRegistrationNumber(), newIncorrectDate);
+
+    }
+
+    @Test
+    public void testRemovingAllItems() throws Exception {
```

Comment:
testujesz usuwanie, którego nawet nie wykorzystujesz w kodzie produkcyjnym? po co?

Reply:
Zgadzam się, że to bez sensu.